flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/EdgeManagerBuildUtilTest.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.executiongraph;
+
+import org.apache.flink.runtime.io.network.partition.ResultPartitionType;
+import org.apache.flink.runtime.jobgraph.DistributionPattern;
+import org.apache.flink.runtime.jobgraph.JobVertex;
+import org.apache.flink.runtime.jobgraph.tasks.AbstractInvokable;
+import org.apache.flink.runtime.scheduler.SchedulerBase;
+import org.apache.flink.runtime.scheduler.strategy.ConsumerVertexGroup;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.apache.flink.runtime.jobgraph.DistributionPattern.ALL_TO_ALL;
+import static org.apache.flink.runtime.jobgraph.DistributionPattern.POINTWISE;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests for {@link EdgeManagerBuildUtil} to verify the max number of connecting edges between
+ * vertices for pattern of both {@link DistributionPattern#POINTWISE} and {@link
+ * DistributionPattern#ALL_TO_ALL}.
+ */
+public class EdgeManagerBuildUtilTest {
+
+    @Test
+    public void testPointwise() throws Exception {
+        test(17, 17, POINTWISE);
+        test(17, 23, POINTWISE);
+        test(17, 34, POINTWISE);
+        test(34, 17, POINTWISE);
+        test(23, 17, POINTWISE);
+    }
+
+    @Test
+    public void testAllToAll() throws Exception {
+        test(17, 17, ALL_TO_ALL);
+        test(17, 23, ALL_TO_ALL);
+        test(17, 34, ALL_TO_ALL);
+        test(34, 17, ALL_TO_ALL);
+        test(23, 17, ALL_TO_ALL);
+    }
+
+    private void test(int upstream, int downstream, DistributionPattern pattern) throws Exception {
```

Comment:
I think the name should be `testGetMaxNumEdgesToTarget()` to be easier to understand what it does.
Furthermore, `testPointwise` can be `testGetMaxNumEdgesToTargetInPointwiseConnection`, etc.

Reply:
Updated